src/webapi/AppSettings.cs

```diff
@@ -54,6 +55,30 @@ public static string[] AllowedOrigins
             return GetInstance()._configuration.GetValue<string>("CorsAllowedOrigins").Split(",");
         }
     }
+
+    public static string BasePath
+    {
+        get
+        {
+            return GetInstance()._configuration.GetValue<string>("BasePath");
+        }
+    }
+
+    // Add a method to accept all SSL certs if the TrustAllSSLCerts is true in the appsettings.json file.
+    public static void TrustAllSSLCerts()
+    {
+        if (GetInstance()._configuration.GetValue<bool>("TrustAllSSLCerts"))
+        {
+            // Trust all SSL certs -- needed unless signed SSL certificates are configured.
+            System.Net.ServicePointManager.ServerCertificateValidationCallback =
+                ((sender, certificate, chain, sslPolicyErrors) => true);
+
+            // Due to our reliance on some older libraries, the.NET framework won't necessarily default
+            // to the latest unless we explicitly request it. Some hosting environments will not allow older versions
+            // of TLS, and thus calls can fail without this extra configuration.
+            System.Net.ServicePointManager.SecurityProtocol |= System.Net.SecurityProtocolType.Tls11 | System.Net.SecurityProtocolType.Tls12;
+        }
+    }
```

Comment:
For right now, let's leave this code alone. My comments are just educational.

There is a concept called the Single Responsibility Principle (SRP), sometimes also called "do just one thing". I have a [blog post](https://blog.safnet.com/best-practices-tdd-oo/solid-testing) that addresses SRP and related topics, which you might find interesting.

In this case, the `AppSettings` file's _single responsibility_ should be to retrieve configuration values. This function does more than that: it also makes modifications to the security.

In the long term, it would be better for those modifications to occur in `Program.cs`, whose single responsibility is to setup the application infrastructure. Then in `AppSettings` you would just have:

```
    public static bool TrustAllSSLCerts
    {
        get
        {
            return GetInstance()._configuration.GetValue<bool>("TrustAllSSLCerts");
        }
    }
```

And then in `Program.cs` you would have

```
   if (AppSettings.TrustAllSSLCerts)
        {
            // Trust all SSL certs -- needed unless signed SSL certificates are configured.
            System.Net.ServicePointManager.ServerCertificateValidationCallback =
                ((sender, certificate, chain, sslPolicyErrors) => true);

            // Due to our reliance on some older libraries, the.NET framework won't necessarily default
            // to the latest unless we explicitly request it. Some hosting environments will not allow older versions
            // of TLS, and thus calls can fail without this extra configuration.
            System.Net.ServicePointManager.SecurityProtocol |= System.Net.SecurityProtocolType.Tls11 | System.Net.SecurityProtocolType.Tls12;
        }
```

You can refactor this in your next ticket. Let's leave it alone right now.